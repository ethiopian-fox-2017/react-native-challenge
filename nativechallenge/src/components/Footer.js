import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  Button
} from 'react-native'

const styles = StyleSheet.create({
  footer: {
    width: '100%',
    height: '10%',
    backgroundColor: '#80CBC4',
    justifyContent: 'flex-start',
    borderTopWidth: 1
  },
  footerContainer: {
    flex: 2,
    flexDirection: 'row',
    justifyContent: 'space-around',
  },
  btnFooter: {
    width: '20%',
  }
});

const Footer = () => {
  return (
    <View style = {styles.footer}>

      <View style = {styles.footerContainer}>
        <Button
          onPress={() => {}}
          title="Home"
          color="#000"
          accessibilityLabel="Learn more about this purple button"
          style = {styles.btnFooter }
        />
        <Button
          onPress={() => {}}
          title="About"
          color="#000"
          accessibilityLabel="Learn more about this purple button"
          style = {styles.btnFooter }
        />
        <Button
          onPress={() => {}}
          title="Chat"
          color="#000"
          accessibilityLabel="Learn more about this purple button"
          style = {styles.btnFooter }
        />
        <Button
          onPress={() => {}}
          title="Profile"
          color="#000"
          accessibilityLabel="Learn more about this purple button"
          style = {styles.btnFooter }
        />
      </View>

    </View>
  )

}

export default Footer;
