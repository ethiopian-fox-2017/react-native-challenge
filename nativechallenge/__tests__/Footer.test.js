import 'react-native';
import React from 'react';
import Footer from '../src/components/Footer.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

it('renders correctly', () => {
  const tree = shallow(
    <Footer />
  );
  expect(tree).toMatchSnapshot()
});
