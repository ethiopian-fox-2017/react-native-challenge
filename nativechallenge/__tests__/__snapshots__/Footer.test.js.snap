// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Object {
            "backgroundColor": "#80CBC4",
            "borderTopWidth": 1,
            "height": "10%",
            "justifyContent": "flex-start",
            "width": "100%",
          }
    }
>
    <View
        style={
            Object {
                "flex": 2,
                "flexDirection": "row",
                "justifyContent": "space-around",
              }
        }
    >
        <Button
            accessibilityLabel="Learn more about this purple button"
            color="#000"
            onPress={[Function]}
            style={
                Object {
                    "width": "20%",
                  }
            }
            title="Home"
        />
        <Button
            accessibilityLabel="Learn more about this purple button"
            color="#000"
            onPress={[Function]}
            style={
                Object {
                    "width": "20%",
                  }
            }
            title="About"
        />
        <Button
            accessibilityLabel="Learn more about this purple button"
            color="#000"
            onPress={[Function]}
            style={
                Object {
                    "width": "20%",
                  }
            }
            title="Chat"
        />
        <Button
            accessibilityLabel="Learn more about this purple button"
            color="#000"
            onPress={[Function]}
            style={
                Object {
                    "width": "20%",
                  }
            }
            title="Profile"
        />
    </View>
</View>,
  "nodes": Array [
    <View
      style={
            Object {
                  "backgroundColor": "#80CBC4",
                  "borderTopWidth": 1,
                  "height": "10%",
                  "justifyContent": "flex-start",
                  "width": "100%",
                }
      }
>
      <View
            style={
                  Object {
                        "flex": 2,
                        "flexDirection": "row",
                        "justifyContent": "space-around",
                      }
            }
      >
            <Button
                  accessibilityLabel="Learn more about this purple button"
                  color="#000"
                  onPress={[Function]}
                  style={
                        Object {
                              "width": "20%",
                            }
                  }
                  title="Home"
            />
            <Button
                  accessibilityLabel="Learn more about this purple button"
                  color="#000"
                  onPress={[Function]}
                  style={
                        Object {
                              "width": "20%",
                            }
                  }
                  title="About"
            />
            <Button
                  accessibilityLabel="Learn more about this purple button"
                  color="#000"
                  onPress={[Function]}
                  style={
                        Object {
                              "width": "20%",
                            }
                  }
                  title="Chat"
            />
            <Button
                  accessibilityLabel="Learn more about this purple button"
                  color="#000"
                  onPress={[Function]}
                  style={
                        Object {
                              "width": "20%",
                            }
                  }
                  title="Profile"
            />
      </View>
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Footer />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Object {
                              "backgroundColor": "#80CBC4",
                              "borderTopWidth": 1,
                              "height": "10%",
                              "justifyContent": "flex-start",
                              "width": "100%",
                            }
          }
>
          <View
                    style={
                              Object {
                                        "flex": 2,
                                        "flexDirection": "row",
                                        "justifyContent": "space-around",
                                      }
                    }
          >
                    <Button
                              accessibilityLabel="Learn more about this purple button"
                              color="#000"
                              onPress={[Function]}
                              style={
                                        Object {
                                                  "width": "20%",
                                                }
                              }
                              title="Home"
                    />
                    <Button
                              accessibilityLabel="Learn more about this purple button"
                              color="#000"
                              onPress={[Function]}
                              style={
                                        Object {
                                                  "width": "20%",
                                                }
                              }
                              title="About"
                    />
                    <Button
                              accessibilityLabel="Learn more about this purple button"
                              color="#000"
                              onPress={[Function]}
                              style={
                                        Object {
                                                  "width": "20%",
                                                }
                              }
                              title="Chat"
                    />
                    <Button
                              accessibilityLabel="Learn more about this purple button"
                              color="#000"
                              onPress={[Function]}
                              style={
                                        Object {
                                                  "width": "20%",
                                                }
                              }
                              title="Profile"
                    />
          </View>
</View>,
        "_debugID": 2,
        "_renderedOutput": <View
          style={
                    Object {
                              "backgroundColor": "#80CBC4",
                              "borderTopWidth": 1,
                              "height": "10%",
                              "justifyContent": "flex-start",
                              "width": "100%",
                            }
          }
>
          <View
                    style={
                              Object {
                                        "flex": 2,
                                        "flexDirection": "row",
                                        "justifyContent": "space-around",
                                      }
                    }
          >
                    <Button
                              accessibilityLabel="Learn more about this purple button"
                              color="#000"
                              onPress={[Function]}
                              style={
                                        Object {
                                                  "width": "20%",
                                                }
                              }
                              title="Home"
                    />
                    <Button
                              accessibilityLabel="Learn more about this purple button"
                              color="#000"
                              onPress={[Function]}
                              style={
                                        Object {
                                                  "width": "20%",
                                                }
                              }
                              title="About"
                    />
                    <Button
                              accessibilityLabel="Learn more about this purple button"
                              color="#000"
                              onPress={[Function]}
                              style={
                                        Object {
                                                  "width": "20%",
                                                }
                              }
                              title="Chat"
                    />
                    <Button
                              accessibilityLabel="Learn more about this purple button"
                              color="#000"
                              onPress={[Function]}
                              style={
                                        Object {
                                                  "width": "20%",
                                                }
                              }
                              title="Profile"
                    />
          </View>
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Footer />,
}
`;
